receiver/etwreceiver/receiver_windows.go

```diff
@@ -183,6 +222,34 @@ func (r *etwReceiver) Start(ctx context.Context, _ component.Host) error {
 	return err
 }
 
+// processLogs handles the log consumption in separate goroutines
+func (r *etwReceiver) processLogs(ctx context.Context) {
+	defer r.wg.Done()
+
+	for {
+		select {
+		case eventData, ok := <-r.eventChan:
+			if !ok {
+				// Channel closed, exit worker
+				return
+			}
+
+			r.logger.Debug("Consuming logs")
+			r.obsrecv.StartLogsOp(ctx)
+			count := eventData.LogRecordCount()
+			err := r.logsConsumer.ConsumeLogs(ctx, *eventData)
+			if err != nil {
+				r.logger.Error("Failed to consume logs", zap.Error(err))
+			}
+			r.obsrecv.EndLogsOp(ctx, reportFormat, count, err)
+
+		case <-ctx.Done():
+			// Context cancelled, exit worker
+			return
+		}
```

Comment:
Check cancellation first as it is higher priority.

```suggestion
	for {
		select {
		case <-ctx.Done():
			// Context cancelled, exit worker
			return
		case eventData, ok := <-r.eventChan:
			if !ok {
				// Channel closed, exit worker
				return
			}

			r.logger.Debug("Consuming logs")
			r.obsrecv.StartLogsOp(ctx)
			count := eventData.LogRecordCount()
			err := r.logsConsumer.ConsumeLogs(ctx, *eventData)
			if err != nil {
				r.logger.Error("Failed to consume logs", zap.Error(err))
			}
			r.obsrecv.EndLogsOp(ctx, reportFormat, count, err)
		}
```